run_tests.py

```
@@ -252,20 +254,28 @@ def run_tests(self) -> bool:
+        # modules like ragflow/deepdoc). Toggle with RAGFLOW_TEST_IMPORT_TRACE=1.
+        final_cmd = cmd
+        if os.environ.get("RAGFLOW_TEST_IMPORT_TRACE") == "1":
+            final_cmd = [self.python, "-v", "-m", "pytest", *cmd[1:]]
+
+        print(f"\n{set_color(s='[EXECUTING]', color='blue')} {' '.join(final_cmd)}\n", flush=True)
+        self.print_info("Launching pytest — collection of test modules may take a while...")
 
         # Run pytest
         try:
-            result = subprocess.run(cmd, check=False)
+            result = subprocess.run(final_cmd, check=False)
```

Ruff (0.16.0)

[error] `subprocess` call: check for execution of untrusted input

(S603)